Userland/Libraries/LibJS/Runtime/AtomicsObject.cpp

```diff
@@ -4,11 +4,132 @@
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
+#include <AK/Atomic.h>
+#include <AK/ByteBuffer.h>
 #include <LibJS/Runtime/AtomicsObject.h>
 #include <LibJS/Runtime/GlobalObject.h>
+#include <LibJS/Runtime/TypedArray.h>
+#include <LibJS/Runtime/Value.h>
 
 namespace JS {
 
+static TypedArrayBase* typed_array_from(GlobalObject& global_object, Value typed_array_value)
+{
+    auto& vm = global_object.vm();
+
+    auto* this_object = typed_array_value.to_object(global_object);
+    if (!this_object)
+        return nullptr;
+    if (!this_object->is_typed_array()) {
+        vm.throw_exception<TypeError>(global_object, ErrorType::NotA, "TypedArray");
+        return nullptr;
+    }
+
+    return static_cast<TypedArrayBase*>(this_object);
+}
+
+// 23.2.4.3 ValidateTypedArray ( O ), https://tc39.es/ecma262/#sec-validatetypedarray
+static void validate_typed_array(GlobalObject& global_object, TypedArrayBase& typed_array)
+{
+    auto& vm = global_object.vm();
+
+    if (!typed_array.is_typed_array())
+        vm.throw_exception<TypeError>(global_object, ErrorType::NotA, "TypedArray");
+    else if (typed_array.viewed_array_buffer()->is_detached())
+        vm.throw_exception<TypeError>(global_object, ErrorType::DetachedArrayBuffer);
+}
```

Comment:
These don't belong here - let's move them to TypedArray.{h,cpp} and replace the ones in TypedArrayConstructor.